salt/minion.py

```diff
@@ -1253,6 +1253,7 @@ def _post_master_init(self, master):
                     'minutes': self.opts['mine_interval'],
                     'jid_include': True,
                     'maxrunning': 2,
+                    'run_on_start': True,
```

Comment:
have you tested that this works? because my assumption is this would only run on the minion start, and as this is possibly being added after the minion start, i am not sure that it is going to do anything.

Reply:
i compared two minion starts, one with that change and one without it. it seems like the mine.update is executed right after being added

without change:
```
[TRACE   ] ext_pillar_extra_data = {}
[TRACE   ] ==== evaluating schedule now None =====
[DEBUG   ] salt.crypt.get_rsa_key: Loading private key
```

with change:
```
[TRACE   ] ext_pillar_extra_data = {}
[TRACE   ] ==== evaluating schedule now None =====
[INFO    ] Running scheduled job: __mine_interval
[DEBUG   ] schedule: This job was scheduled with jid_include, adding to cache (jid_include defaults to True)
[DEBUG   ] schedule: This job was scheduled with a max number of 2
[DEBUG   ] schedule.handle_func: adding this job to the jobcache with data {u'fun_args': [], u'jid': u'20180817202306469000', u'schedule': u'__mine_interval', u'pid': 5188, u'fun': u'mine.update', u'id': u'somename'}
[DEBUG   ] LazyLoaded mine.update
[DEBUG   ] LazyLoaded config.merge
[TRACE   ] Error loading module.network: The network execution module cannot be loaded on Windows: use win_network instead.
[TRACE   ] Loaded win_network as virtual network
[DEBUG   ] LazyLoaded network.ip_addrs
[DEBUG   ] Initializing COM library
[DEBUG   ] Uninitializing COM library
[DEBUG   ] MinionEvent PUB socket URI: 4510
[DEBUG   ] MinionEvent PULL socket URI: 4511
[DEBUG   ] Initializing new IPCClient for path: 4511
[TRACE   ] IPCClient: Connecting to socket: 4511
[TRACE   ] IPCServer: Handling connection to address: ('127.0.0.1', 50773)
[DEBUG   ] Sending event: tag = _minion_mine; data = {u'clear': False, u'cmd': u'_mine', u'data': {'network.ip_addrs': [u'someval']}, u'id': u'somename', u'_stamp': '2018-08-17T20:23:06.609000'}
[DEBUG   ] Minion of 'someip' is handling event tag '_minion_mine'
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'c:\\salt\\conf\\pki\\minion', u'somename', u'tcp://someip:4506', u'aes')
[DEBUG   ] Initializing new AsyncAuth for (u'c:\\salt\\conf\\pki\\minion', u'somename', u'tcp://someip:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://someip:4506
[DEBUG   ] Trying to connect to: tcp://someip:4506
[TRACE   ] Inserted key into loop_instance_map id 128487304 for key (u'c:\\salt\\conf\\pki\\minion', u'somename', u'tcp://someip:4506', u'aes') and process 5188
[DEBUG   ] salt.crypt.get_rsa_key: Loading private key
```

and gets correct values from the network.ip_addrs function

Reply:
btw the minion.d/_schedule.conf was manually emptied before the run with the change to make ensure it's like a fresh start